clients/src/main/java/org/apache/kafka/clients/consumer/channels/KafkaServerEventQueue.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.consumer.channels;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.events.KafkaServerEvent;
+
+import java.util.Optional;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+
+public class KafkaServerEventQueue {
```

Comment:
I'm not sure that I see the value just yet of a separate class for the queue. Mostly because it limits flexibility to use the collection in utility methods.

Maybe there will be more logic added to it later?

Reply:
I think in the future we might need to add generation logic, etc. to this class

Reply:
Is it not a bit odd to have a client class with the `KafkaServer` prefix? Also, does the `Kafka` prefix help given that all this code is for Kafka?